classes/notification/managerDelegate/PendingRevisionsNotificationManager.inc.php

```diff
@@ -33,19 +33,11 @@ public function getNotificationUrl($request, $notification) {
 		$submissionDao = Application::getSubmissionDAO();
 		$submission = $submissionDao->getById($notification->getAssocId());
 
-		import('lib.pkp.controllers.grid.submissions.SubmissionsListGridCellProvider');
-		list($page, $operation) = SubmissionsListGridCellProvider::getPageAndOperationByUserRoles($request, $submission, $notification->getUserId());
-
-		if ($page == 'workflow') {
-			$stageData = $this->_getStageDataByType();
-			$operation = $stageData['path'];
-		}
+		$stageData = $this->_getStageDataByType();
+		$operation = $stageData['path'];
 
-		$router = $request->getRouter();
-		$dispatcher = $router->getDispatcher();
-		$contextDao = Application::getContextDAO();
-		$context = $contextDao->getById($submission->getContextId());
-		return $dispatcher->url($request, ROUTE_PAGE, $context->getPath(), $page, $operation, $submission->getId());
+		import('lib.pkp.controllers.grid.submissions.SubmissionsListGridCellProvider');
+		return SubmissionsListGridCellProvider::getUrlByUserRoles($request, $submission, $notification->getUserId(), $stageData['path']);
```

Comment:
I think you can use $operation instead of the harder way $stageData['path'] right?